pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -254,4 +254,31 @@ func AddAlertmanagerConfigMigrations(mg *migrator.Migrator) {
 	mg.AddMigration("Add column default in alert_configuration", migrator.NewAddColumnMigration(alertConfiguration, &migrator.Column{
 		Name: "default", Type: migrator.DB_Bool, Nullable: false, Default: "0",
 	}))
+
+	//------- rename table ------------------
+	tempTable := "alert_configuration_TBD" // do not change it
+	migrationId := fmt.Sprintf("Rename table %s to %s - v1", alertConfiguration.Name, tempTable)
+	mg.AddMigration(migrationId, migrator.NewRenameTableMigration(alertConfiguration.Name, tempTable))
+
+	alertConfigurationV2 := migrator.Table{
+		Name: "alert_configuration",
+		Columns: []*migrator.Column{
+			{Name: "id", Type: migrator.DB_BigInt, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "alertmanager_configuration", Type: migrator.DB_Text, Nullable: false},
+			{Name: "configuration_version", Type: migrator.DB_NVarchar, Length: 3}, // In a format of vXX e.g. v1, v2, v10, etc
+			{Name: "created_at", Type: migrator.DB_DateTime, Nullable: false, Default: time.Now().Format("2006-01-02 15:04:05")},
```

Comment:
Overall, the changes LGTM but I'm not sure about this type change for MySQL.

According to the xorm documentation, this is supported:

```
type User struct {
    Id int64
    Name string
    CreatedAt int64 `xorm:"created"`
}
```

Could it be that there's a bug in the MySQL translation in xorm? Either way, the problem here is that the format of the date is incomplete, yet the created at is not used for anything at the moment - so it might not be worth fixing?

Reply:
In our case [the struct field is time.Time](https://github.com/grafana/grafana/blob/13cf6a3e00ccb4132e7005c1abda3f053c0b9af1/pkg/services/ngalert/models/alertmanager.go#L13) but the database type is int and that's why I believe that `xorm` get confused.

Also, the problem if we don't fix is that it fails completely for PostgreSQL because `xorm` sends a string value:
`pq: invalid input syntax for integer: \"2021-06-02 10:48:23\`

In any case, it would be better if we didn't let `xorm` handle this but we would provide a value ourselves.